src/AppBundle/Entity/SocialShare.php

```diff
@@ -0,0 +1,227 @@
+<?php
+
+namespace AppBundle\Entity;
+
+use AppBundle\Utils\EmojisRemover;
+use Doctrine\ORM\Mapping as ORM;
+use Gedmo\Mapping\Annotation as Gedmo;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="social_shares")
+ */
+class SocialShare
+{
+    const TYPE_IMAGE = 'image';
+    const TYPE_VIDEO = 'video';
+    const TYPE_PDF = 'pdf';
+
+    const TYPES = [
+        self::TYPE_IMAGE,
+        self::TYPE_VIDEO,
+        self::TYPE_PDF,
+    ];
+
+    use EntityTimestampableTrait;
+
+    /**
+     * @ORM\Column(type="bigint")
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    private $id;
+
+    /**
+     * @ORM\Column(length=100)
+     *
+     * @Assert\NotBlank
+     */
+    private $name;
+
+    /**
+     * @ORM\Column
+     *
+     * @Gedmo\Slug(fields={"name"})
+     */
+    private $slug;
+
+    /**
+     * @ORM\Column(length=10)
+     *
+     * @Assert\NotBlank
+     * @Assert\Choice({"image", "video", "pdf"})
+     */
+    private $type;
+
+    /**
+     * @ORM\Column(type="integer")
+     */
+    private $position;
+
+    /**
+     * @ORM\Column(type="text")
+     *
+     * @Assert\NotBlank
+     */
+    private $description;
+
+    /**
+     * @ORM\Column
+     *
+     * @Assert\NotBlank
+     * @Assert\Url
+     */
+    private $defaultUrl;
+
+    /**
+     * @ORM\Column(nullable=true)
+     *
+     * @Assert\Url
+     */
+    private $facebookUrl;
+
+    /**
+     * @ORM\Column(nullable=true)
+     *
+     * @Assert\Url
+     */
+    private $twitterUrl;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\SocialShareCategory")
+     *
+     * @Assert\NotBlank
+     */
+    private $socialShareCategory;
+
+    /**
+     * @ORM\ManyToOne(targetEntity="AppBundle\Entity\Media")
+     *
+     * @Assert\NotBlank
+     */
+    private $media;
+
+    /**
+     * @ORM\Column(type="boolean")
+     */
+    private $published;
+
+    public function __construct(string $name = '', int $position = 1, $published = false)
+    {
+        $this->name = EmojisRemover::remove($name);
+        $this->position = $position;
+        $this->published = $published;
+        $this->type = self::TYPE_IMAGE;
+    }
+
+    public function getId(): int
+    {
+        return $this->id;
+    }
+
+    public function getName(): ?string
+    {
+        return $this->name;
+    }
+
+    public function setName(string $name)
```

Comment:
Yes but the form should be able to put null into it and handle the problem with the validator.